### PR TITLE
Feature resize board

### PR DESCRIPTION
**Pull request by melonmanchan**
I am sorry

### CODE REVIEW

**Comment by n4sjamk-docker** [comment]
Can one of the admins verify this patch?

File: routes/board.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
 var _        = require('lodash');
 var express  = require('express');
 var mongoose = require('mongoose');
+var Promise  = require('promise')
```
Comment (tanelih) [inline routes/board.js]: Don't trust ASI.

File: routes/board.js [hunk 2]
```diff
@@ -152,6 +153,21 @@ Router.route('/boards/:board_id')
 				return next(utils.error(400, err));
 			}
 			Board.populate(board, 'createdBy', function(err, board) {
+
+				if(req.resolved.board.size.width < old.size.width || req.resolved.board.size.height < old.size.height){
+					Ticket.find({ 'board': req.resolved.board.id,
+						$or: [
+								{'position.x': {$gt: (req.resolved.board.size.width * 192) - 96}},
```
Comment (tanelih) [inline routes/board.js]: What are these magic numbers? Maybe declare them as variables as to make the code more readable.

File: routes/board.js [hunk 3]
```diff
@@ -152,6 +153,21 @@ Router.route('/boards/:board_id')
...
+								{'position.y': {$gt: (req.resolved.board.size.height * 108) - 54}}
```
Comment (tanelih) [inline routes/board.js]: Magic numbers!

File: routes/board.js [hunk 4]
```diff
@@ -152,6 +153,21 @@ Router.route('/boards/:board_id')
...
+						     ]}, function (err, tickets) {
+
+						if(tickets.length > 0) {
+							Promise.all(tickets.map(utils.ticketClamper(req.resolved.board))).then(function(){
```
Comment (tanelih) [inline routes/board.js]: Why do we have an empty `then`?

File: utils/clamper.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+var mongoose = require('mongoose');
+var emitter  = require('./emitter');
+var Promise  = require('promise');
+
+var Event    = mongoose.model('event', require('../config/schemas/event'));
```
Comment (tanelih) [inline utils/clamper.js]: Too much space.

File: routes/board.js [hunk 6]
```diff
@@ -152,6 +153,21 @@ Router.route('/boards/:board_id')
 				return next(utils.error(400, err));
 			}
 			Board.populate(board, 'createdBy', function(err, board) {
+
+				if(req.resolved.board.size.width < old.size.width || req.resolved.board.size.height < old.size.height){
+					Ticket.find({ 'board': req.resolved.board.id,
+						$or: [
+								{'position.x': {$gt: (req.resolved.board.size.width * 192) - 96}},
+								{'position.y': {$gt: (req.resolved.board.size.height * 108) - 54}}
+						     ]}, function (err, tickets) {
```
Comment (tanelih) [inline routes/board.js]: Error handling...

File: routes/board.js [hunk 7]
```diff
@@ -152,6 +153,21 @@ Router.route('/boards/:board_id')
...
+
+						if(tickets.length > 0) {
+							Promise.all(tickets.map(utils.ticketClamper(req.resolved.board))).then(function(){
+							})
+						}
+					});
+				}
+
 				if(err) {
```
Comment (tanelih) [inline routes/board.js]: This should probably be about the `populate`.

File: utils/clamper.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+                var oldPosition = {x: ticket.position.x, y: ticket.position.y};
+
+                if(ticket.position.x > (board.size.width * 192) - 96){
+                    ticket.position.x = (board.size.width * 192) - 192;
+                }
+
+                if(ticket.position.y > (board.size.height * 108) - 54){
+                    ticket.position.y = (board.size.height * 108) - 108;
+                }
+
```
Comment (tanelih) [inline utils/clamper.js]: Magic numbers...

File: utils/clamper.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                ticket.save(function(err) {
+                    // we've saved the dog into the db here
```
Comment (tanelih) [inline utils/clamper.js]: This is dog.

File: utils/clamper.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+                        }
+                    }).save(function(err, ev) {
+                            if(err) {
+                                return console.error(err);
+                            }
+                            emitter.to(ev.board)
+                                .emit('board:event', ev.toObject());
+                            return resolve();
+                        });
+
```
Comment (tanelih) [inline utils/clamper.js]: What is this no newline shenanigans?

File: utils/clamper.js [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+                });
+
+            });
+        }
+}
```
Comment (tanelih) [inline utils/clamper.js]: Newline at the end of the line because burger.

File: utils/index.js [hunk 12]
```diff
@@ -2,3 +2,4 @@
 
 module.exports.error   = require('./error');
 module.exports.emitter = require('./emitter');
+module.exports.ticketClamper = require('./clamper');
```
Comment (tanelih) [inline utils/index.js]: Align stuff please. :hamburger:

**Comment by tanelih** [comment]
Good Stuff! :melon:

**Pull request merged**